File: data/readable_format/fortran/file868.f90
module BoxMullerMod
    implicit none
contains
    subroutine boxMuller(x, y)
        real, intent(out) :: x, y
        real :: u1, u2, r, theta
        call random_seed()
        call random_number(u1)
        call random_number(u2)
        r = sqrt(-2.0 * log(u1))
        theta = 2.0 * 3.14159265358979323846 * u2
        x = r * cos(theta)
        y = r * sin(theta)
    end subroutine boxMuller
end module BoxMullerMod

program test_box_muller
    use BoxMullerMod
    implicit none
    real :: x, y, mean, variance, start, finish
    integer :: i, Nsim

    Nsim = 10000
    mean = 0.0
    variance = 0.0

    call cpu_time(start)
    do i = 1, Nsim
        call boxMuller(x, y)
        mean = mean + x + y
        variance = variance + x**2 + y**2
    end do
    call cpu_time(finish)

    mean = mean / (2*Nsim)
    variance = variance / (2*Nsim) - mean**2

    print *, "Mean:", mean
    print *, "Variance:", variance
    print *, "Elapsed Time:", finish - start, "seconds"
end program test_box_muller

File: data/readable_format/cpp/file868.cpp
#include <cmath>
#include <chrono>
#include <random>
#include <iostream>

std::pair<double, double> boxMuller() {
    static std::random_device rd;
    static std::mt19937 gen(rd());
    static std::uniform_real_distribution<> dis(0.0, 1.0);

    double u1 = dis(gen);
    double u2 = dis(gen);
    double r = sqrt(-2.0 * log(u1));
    double theta = 2.0 * M_PI * u2;
    double x = r * cos(theta);
    double y = r * sin(theta);

    return std::make_pair(x, y);
}

int main() {
    const int Nsim = 10000;
    double mean = 0.0, variance = 0.0;
    auto start = std::chrono::high_resolution_clock::now();

    for (int i = 0; i < Nsim; ++i) {
        auto nums = boxMuller();
        mean += nums.first + nums.second;
        variance += nums.first * nums.first + nums.second * nums.second;
    }

    auto finish = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = finish - start;
    
    mean /= (2*Nsim);
    variance /= (2*Nsim);
    variance -= mean * mean;

    std::cout << "Mean: " << mean << std::endl;
    std::cout << "Variance: " << variance << std::endl;
    std::cout << "Elapsed Time: " << elapsed.count() << " seconds" << std::endl;

    return 0;
}